systemd/emulation.go

```diff
@@ -98,12 +99,18 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", fmt.Errorf("bind-mounted directory is not supported in emulation mode")
 	}
 
-	mountUnitName, actualFsType, options, err := writeMountUnitFile(snapName, revision, what, where, fstype)
+	// In emulation mode actualFsType is the fs we want to use to manually mount
+	// the snap below, but fstype is used for the created mount unit.
+	// This means that when preseeding in a lxd container, the snap will be
+	// mounted with fuse, but mount unit will use squashfs.
+	mountUnitOptions := append(fsMountOptions(fstype), squashfs.StandardOptions()...)
+	mountUnitName, err := writeMountUnitFile(snapName, revision, what, where, fstype, mountUnitOptions)
```

Comment:
Wonder whether we should provide a list of requirements for the target execution environment that gets set up by preseeding. Right now, we assume that the target system supports squashfs. If I understand correctly, there is no automatic fallback in systemd/mount that would use squashfuse when the kernel driver is unavailable.

Reply:
Yes, targets need to support squashfs, preseeding was implemented with speeding of firstboot of ubuntu cloud images in mind. We don't pretend we run elsewhere and therefore exclude snap-preseed binary from other systems. The set of fixes I've proposed makes snap-preseed usable from lxd container, but the requirements for target system don't change.